### ISSUE TITLE
Issue with bootstapping version 2.9.0

### ISSUE THREAD

**Issue by SethosII** [hpcugent/easybuild-framework#2004]
I trried to install EasyBuild 2.9.0 by bootstrapping from source tarballs. Therefore I downloaded the `bootstrap_eb.py`, `easybuild-easyblocks-`, `-framwork-` and `-easyconfig-v-2.9.0.tar.gz` from the github repositories.

But bootstrapping fails during the sanity check because no module is created:

```
> cd /easybuild     
> ls bootstrap/
bootstrap_eb.py
easybuild-easyblocks-v2.9.0.tar.gz
easybuild-framework-v2.9.0.tar.gz
easybuild-easyconfigs-v2.9.0.tar.gz
> export EASYBUILD_BOOTSTRAP_SOURCEPATH=/easybuild/bootstrap/ 
> ./bootstrap/bootstrap_eb.py .                           
[[INFO]] EasyBuild bootstrap script (version 20161104.01, MD5: 17df92070abdc994a0b2aac2e6cd1f71)
[[INFO]] Found Python 2.7.6 (default, Jun 22 2015, 17:58:13) ; [GCC 4.8.2]

[[INFO]] Installation prefix /easybuild
[[INFO]] Fetching sources from /easybuild/bootstrap/...
[[INFO]] Found module command '/tools/lmod/lmod/libexec/lmod' via $LMOD_CMD (Lmod), so using it.
[[INFO]] No suitable setuptools installation found, proceeding with stage 0...


[[INFO]] +++ STAGE 0: installing distribute via included (patched) distribute_setup.py...

Downloading http://hpcugent.github.io/easybuild/files/distribute-0.6.49-patched1.tar.gz
Extracting in /tmp/tmpqcVeQF
Now working in /tmp/tmpqcVeQF/distribute-0.6.49
Installing Distribute
[[INFO]] Installed setuptools version 0.6 (/tmp/tmpaA4NyS/lib/python2.7/site-packages/distribute-0.6.49-py2.7.egg/setuptools/__init__.pyc)


[[INFO]] +++ STAGE 1: installing EasyBuild in temporary dir with easy_install...

[[INFO]] Fetching sources from /easybuild/bootstrap/...
[[INFO]] Found /easybuild/bootstrap/easybuild-framework-v2.9.0.tar.gz for easybuild-framework package
[[INFO]] Found /easybuild/bootstrap/easybuild-easyblocks-v2.9.0.tar.gz for easybuild-easyblocks package
[[INFO]] Found /easybuild/bootstrap/easybuild-easyconfigs-v2.9.0.tar.gz for easybuild-easyconfigs package
[[INFO]] installing EasyBuild with 'easy_install --quiet --upgrade --prefix=/tmp/tmpaA4NyS/eb_stage1 /easybuild/bootstrap/easybuild-framework-v2.9.0.tar.gz /easybuild/bootstrap/easybuild-easyblocks-v2.9.0.tar.gz /easybuild/bootstrap/easybuild-easyconfigs-v2.9.0.tar.gz'


[[INFO]] +++ STAGE 2: installing EasyBuild in /easybuild with EasyBuild from stage 1...

== temporary log file in case of crash /tmp/eb-jsegd8/easybuild-MrHQYQ.log
== processing EasyBuild easyconfig /tmp/tmpaA4NyS/EasyBuild-2.9.0.eb
== building and installing EasyBuild/2.9.0...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== building...
== testing...
== installing...
== taking care of extensions...
== postprocessing...
== sanity checking...
== FAILED: Installation ended unsuccessfully (build directory: /tmp/tmpaA4NyS/EasyBuild/2.9.0/dummy-dummy): build failed (first 300 chars): Sanity check failed: EasyBuild failed to install, cmd '/usr/bin/python -c "import easybuild"' (stdin: None) output: Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named easybuild
, sanity check command eb --version exited with code 127 (output: /bin/
== Results of the build can be found in the log file(s) /tmp/eb-jsegd8/easybuild-EasyBuild-2.9.0-20161118.074410.KdBRZ.log
Traceback (most recent call last):
  File "./bootstrap/bootstrap_eb.py", line 958, in <module>
    main()
  File "./bootstrap/bootstrap_eb.py", line 761, in main
    stage2(tmpdir, templates, install_path, distribute_egg_dir, sourcepath)
  File "./bootstrap/bootstrap_eb.py", line 661, in stage2
    easybuild_main()
  File "/tmp/tmpaA4NyS/eb_stage1/lib/python2.7/site-packages/easybuild_framework-2.9.0-py2.7.egg/easybuild/main.py", line 411, in main
    ecs_with_res = build_and_install_software(ordered_ecs, init_session_state, exit_on_failure=exit_on_failure)
  File "/tmp/tmpaA4NyS/eb_stage1/lib/python2.7/site-packages/easybuild_framework-2.9.0-py2.7.egg/easybuild/main.py", line 150, in build_and_install_software
    raise EasyBuildError(test_msg)
easybuild.tools.build_log.EasyBuildError: 'Build of /tmp/tmpaA4NyS/EasyBuild-2.9.0.eb failed (err: \'build failed (first 300 chars): Sanity check failed: EasyBuild failed to install, cmd \\\'/usr/bin/python -c "import easybuild"\\\' (stdin: None) output: Traceback (most recent call last):\\n  File "<string>", line 1, in <module>\\nImportError: No module named easybuild\\n, sanity check command eb --version exited with code 127 (output: /bin/\')'
```

And indeed there is no easybuild module in the directory. Tried on Ubuntu Server 14.04.5/16.04.1 with Python 2.7.6/2.7.12 respectively (and Lmod 5.7.2 which is the reason for using easybuild 2.9.0).

Am I missing something?

**Comment by SethosII** [comment]
Yet another try succeeded. In the meantime I installed tcl-dev and liblua-dev packages on these machines, but shouldn't they be unnecessary when Lmod is already present?

**Comment by boegel** [comment]
@SethosII I don't think this is related to installing `tcl-dev` or `liblua-dev`, if there would have been a problem with Lmod, you would have seen a different error.

Did you re-download the bootstrap script on the 2nd try?

If you have a way of reproducing this problem, please run the bootstrap with `$EASYBUILD_BOOTSTRAP_DEBUG` defined to get more details.

I can't tell based on the output what would've gone wrong exactly, but it would be interesting to know...

**Comment by boegel** [comment]
@SethosII Any reason you are stuck to Lmod 5.7.2 btw? We indeed did a small bump in the Lmod version requirement, but this was mostly motivated by some code cleanup we could do if we knew Lmod 5.8 or newer was used, see https://github.com/hpcugent/easybuild-framework/pull/1985/files .

If this is a big issue, we can reconsider this and stick to supporting Lmod 5.6.3 and newer...

**Comment by SethosII** [comment]
Hey @boegel, I'm indeed able to reproduce this by installing/uninstalling `tcl-dev` and `liblua-dev` and their dependencies. These are packages that were installed by installing `tcl-dev` and `liblua-dev`: `liblua5.2-0 liblua5.2-dev libreadline-dev libreadline6-dev libtcl8.6 libtinfo-dev libtool-bin pkg-config tcl tcl-dev tcl8.6 tcl8.6-dev zlib1g-dev`. Fiddled around with these packages and found that `tcl` (with `libtcl8.6 tcl tcl8.6`) was the required package. The installation runs fine with these installed. Maybe because the EasyBuild module itself is written in TCL and not Lua?

Here are the files I used and the log file from a failed run with `$EASYBUILD_BOOTSTRAP_DEBUG`: https://dl.dropboxusercontent.com/u/90335570/easybuild-bootstrap.zip (attaching via GitHub didn't work?).

There is no real reason for using Lmod 5.7.2, it is just what was already there. I would prefer you to step forward and cut old branches. Lazy admins shouldn't be supported ;) While I was trying to build a more recent version of Lmod there was an error about missing tcl/liblua so I installed them and tried again to install EasyBuild and it worked with these libraries.

Just for protocol: all on a Ubuntu server 16.04.1.
